### PR TITLE
Prune cluster cves when cluster is removed

### PR DESCRIPTION
**Pull request by md2119**
## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes ~

If any of these don't apply, please comment below.

## Testing Performed
Unit Test:

### CODE REVIEW

File: central/cluster/datastore/datastore.go [hunk 1]
```diff
@@ -293,5 +297,5 @@ func GetTestRocksBleveDataStore(t *testing.T, rocksengine *rocksdbBase.RocksDB,
 		alertStore, namespaceStore, deploymentStore,
 		nodeStore, podStore, secretStore, netFlowStore, netEntityStore,
 		serviceAccountStore, k8sRoleStore, k8sRoleBindingStore, sensorCnxMgr, nil,
-		dacky, clusterRanker, networkBaselineManager)
+		dacky, clusterRanker, networkBaselineManager, nil)
```
Comment (md2119) [inline central/cluster/datastore/datastore.go:300]: `nil` because this is only postgres path. The non-postgres path is handled thru cluster cve manager.

File: central/cluster/datastore/datastore_impl.go [hunk 2]
```diff
@@ -545,7 +548,11 @@ func (ds *datastoreImpl) postRemoveCluster(ctx context.Context, cluster *storage
 	ds.removeK8SRoles(ctx, cluster)
 	ds.removeRoleBindings(ctx, cluster)
 
-	// TODO: Apparently, cluster cves are only cleaned up at next cve fetch cycle.
```
Comment (md2119) [inline central/cluster/datastore/datastore_impl.go:548]: Actually, the cluster cve removal is handled through the fetcher for non-postgres scenario.

File: central/cluster/datastore/datastore.go [hunk 3]
```diff
@@ -128,21 +131,18 @@ func New(
...
 	} else {
 		ds.searcher = search.New(clusterStorage, indexer, graphProvider, clusterRanker)
 	}
-	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
-		sac.AllowFixedScopes(
-			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
-			sac.ResourceScopeKeys(resources.Cluster)))
-	if err := ds.buildIndex(ctx); err != nil {
+
+	if err := ds.buildIndex(sac.WithAllAccess(context.Background())); err != nil {
```
Comment (theencee) [inline central/cluster/datastore/datastore.go:145]: why the change here?

**Review by theencee — APPROVED** [review]

File: central/cluster/datastore/datastore.go (hunk 3 above)
Comment (md2119) [reply central/cluster/datastore/datastore.go:145]: The internal instantiations are completely controlled by us. For instances like building an index on a startup, we really do not benefit from such instantiations. I remember Viswa, working on an issue in the network graph, and the issue was that deployment perm was not included while building the graph.
